app/models/spree/gateway/authorize_net.rb

```diff
@@ -2,14 +2,27 @@ module Spree
   class Gateway::AuthorizeNet < Gateway
     preference :login, :string
     preference :password, :string
+    preference :server, :string, :default => "live"
+
+    # WARNING: Unlike other payment integrations, 'test mode' in Authorize Net's termiminology does not mean use test server
+    # instead, use preferred server set to either 'live' or 'test'
+    # DO NOT TURN TEST MODE TO ON (EVEN IN QA/STAGING ENVIRONMENTS), it will return 0 as transaction ids under all circumstances
+
+    # here, it overloads the setting on the base class, which confusingly defaults to true
+    preference :test_mode, :boolean, :default => false
 
     def provider_class
       ActiveMerchant::Billing::AuthorizeNetGateway
     end
 
     def options_with_test_preference
-      options_without_test_preference.merge(test: self.preferred_test_mode)
+      if !['live','test'].include?(self.preferred_server)
+        ActiveSupport::Deprecation.warn("You must set your preferred server to either 'live' or 'test'")
```

Comment:
I think it would be more appropriate to throw an exception here. If someone has this misconfigured it would not behave as expected.